server/render/test/index.test.js

```diff
@@ -0,0 +1,249 @@
+};
+
+const ssrDisabledContext = {
+	...ssrCompatibleContext,
+	serverSideRender: false,
+};
+
+describe( 'shouldServerSideRender', () => {
+	beforeEach( () => remock() );
+
+	test( 'feature-flag server-side-render should enable SSR (default behavior)', () => {
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+	} );
+
+	test( 'feature-flag server-side-render should disable SSR', () => {
+		remock( { configServerSideRender: false } );
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( false );
+	} );
+
+	test( 'context.serverSideRender should alter the result', () => {
+		expect( shouldServerSideRender( ssrCompatibleContext ) ).toBe( false ); // due to undefined
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+		expect( shouldServerSideRender( ssrDisabledContext ) ).toBe( false );
+	} );
+
+	test( 'context.layout should alter the result', () => {
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+
+		const SsrEnabledContextWithoutLayout = {
+			...ssrEnabledContext,
+			layout: undefined,
+		};
+		expect( shouldServerSideRender( SsrEnabledContextWithoutLayout ) ).toBe( false );
+	} );
+
+	test( 'context.user should alter the result', () => {
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+
+		const ssrEnabledContextWithUser = {
+			...ssrEnabledContext,
+			user: {
+				name: 'hello-world',
+			},
+		};
+		expect( shouldServerSideRender( ssrEnabledContextWithUser ) ).toBe( false );
+	} );
+
+	test( 'isSectionIsomorphic should alter the result', () => {
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+
+		remock( { isSectionIsomorphic: false } );
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( false );
+	} );
+
+	test( 'isDefaultLocale should alter the result', () => {
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+
+		remock( { isDefaultLocale: false } );
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( false );
+	} );
+} );
+
+describe( 'setShouldServerSideRender', () => {
+	test( 'when query is empty, then sets context.serverSideRender to TRUE - and calls next()', () => {
+		const next = jest.fn();
+		const contextWithoutQueryKeys = {
+			query: {},
+		};
+
+		setShouldServerSideRender( contextWithoutQueryKeys, next );
+		expect( contextWithoutQueryKeys.serverSideRender ).toBe( true );
+		expect( next.mock.calls.length ).toBe( 1 );
+	} );
+
+	test( 'when query has values, then sets context.serverSideRender to FALSE - and calls next()', () => {
+		const next = jest.fn();
+		const contextWithQueryKeys = {
+			query: {
+				hello: 'world',
+			},
+		};
+
+		setShouldServerSideRender( contextWithQueryKeys, next );
+		expect( contextWithQueryKeys.serverSideRender ).toBe( false );
+		expect( next.mock.calls.length ).toBe( 1 );
+	} );
+} );
+
+// doing also the setShouldServerSideRenderLogin the test here because it seems adding it on the client side requires more effort
```

Comment:
Hmm, having the test in a totally different directory than the thing it is testing is rather unfortunate. What would it take to move the test to `client/login/test/ssr.js` (where it should live)?

Reply:
IIRC, I had the SSR functions inside the controller before. Running `npm test-client` did end up with plenty of errors when trying to import them from that location, so I had the impression I had to dig deeper into Jest to see what kind of bootstrapping is done before the test.

Since now the tests are in their own ssr.js, moving them to the right location worked like a charm.